experimental/dds/tree2/src/feature-libraries/editable-tree/editableField.ts

```diff
@@ -392,11 +396,33 @@ export class FieldProxyTarget extends ProxyTarget<FieldAnchor> implements Editab
 		);
 	}
 
+	public treeStatus(): TreeStatus {
+		if (this.isFreed()) {
+			return TreeStatus.Deleted;
+		}
+		const fieldAnchor = this.getAnchor();
+		const parentAnchor = fieldAnchor.parent;
+		// If the parentAnchor is undefined it is a detached field.
+		if (parentAnchor === undefined) {
+			return keyAsDetachedField(fieldAnchor.fieldKey) === rootField
+				? TreeStatus.InDocument
+				: TreeStatus.Removed;
+		}
+		const parentAnchorNode = this.context.forest.anchors.locate(parentAnchor);
+
+		// The parentAnchorNode can be undefined in the case that its corresponding PathNode !== Status.Alive.
+		// This means that the parentAnchor points to an invalid node.
+		if (parentAnchorNode === undefined) {
+			return TreeStatus.Deleted;
+		}
+		return treeStatusFromPath(parentAnchorNode);
```

Comment:
The `anchors.locate` method seems to return `undefined` when its corresponding `PathNode` does not have the status `Status.Alive`. I updated the assert statement to use an if statement instead, and return `TreeStatus.Delete` in this case, but I am not entirely sure if this is correct.

Based on the documentation there seems to be 2 other statuses `Status.Disposed` and `Status.Dangling`. `Status.Dangling` seems to correspond to a deleted node. But does `Status.Disposed` mean that it's removed (but invalid)? If that's the case, we may need to distinguish between the two cases, but there seems to be no good way to get the `PathNode` status.

Should an additional method along the lines of `getPathNodeStatus(anchor:Anchor)` be created in `AnchorSet` so that we could make this distinction?

Reply:
It should be impossible for well-formed code to try and look up a disposed AnchorNode, since the only way to look up an AnchorNode is from an anchor, which is a ref count that keeps the AnchorNode from being disposed.

Also most usages (including editable tree) shouldn't see the dangling state either, since they should really free their anchor when they see a delete event since theres no real point to holding onto an anchor after that point.